_documentation/Recipes/InlineIPsec.md

```diff
@@ -0,0 +1,63 @@
+---
+title: "Inline Acceleration - IPsec"
+layout: documentation
+visibility: public
+---
+
+The Inline IPsec Recipe is an application that enables [strongSwan](https://www.strongswan.org/) to use the [Infrastructure Application Inteface](https://ipdk.io/documentation/Interfaces/InfraApp/), specifically the P4Runtime and OpenConfig gRPCs provided by the [Networking Recipe](https://ipdk.io/documentation/Recipes/InfrastructureNetworking/). OpenConfig is used to configure security associations into the infrastructure and P4 is used program how and where the IPsec is inlined. 
+
+![Top Level Components](../img/InlineIPsec_Recipe.svg "Inline IPsec Top Level"){:width="60%"}
+
+# Infrastructure Application Interface #
+
+ - [strongSwan](https://www.strongswan.org/) is used as the IKE control plane.  No code changes are needed in strongSwan.
+ - A strongSwan plug-in is used to convert security association (SA) information into OpenConfig RPC messages to the underlying infrastructure.  The P4 program defines which modes are required in the pipeline, as well as where the IPsec is inserted into the broader virtual networking pipeline:
+   - **Tunnel vs. Transport**: Determines how the original packet is transformed during encryption/decryption as below: 
+![IPsec Packet Transforms](../img/IPsec_Packet_Transforms.svg "IPsec Packet Transforms"){:width="60%"}
+   - **Encrypted Virtual Ports**: Encrypted tunnels between devices over physical ports.  On the wire this wraps additional headers on the outside of the packet in the same way that network virtualization headers such as VXLAN are added/removed.
```

Comment:
I think you have the wording for virtual ports and physical ports backwards here.

Reply:
Whoops!  Thank you I fixed this